Adding devspace files

andresllh opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Adding devspace files for developers that want to use devspace.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Installed [devspace](devspace.sh) cli  
Ran `devspace init` to create devspace_start.sh script and devspace.yaml files
Modified devspace.yaml with correct config for the odh-model-controller.
Ran `devspace dev` and terminal came up.
Changed directory to `/app/cmd` and ran `go run main.go` and the operator started successfully

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

spolti commented:
For this, maybe we could create a new directory like tools, or put it inside hack, would it make sense?

andresllh commented:
/cc @israel-hdez  @mholder6

andresllh commented:
> For this, maybe we could create a new directory like tools, or put it inside hack, would it make sense?

Created `dev_tools` folder that lives at the root of the repository and moved the files in  there. Updated config to reflect the new directory.

israel-hdez reviewed (approved):
Dev tooling :-) 
Let's merge it. I'm not familiar, but this can be useful for people.
/lgtm
